src/actions/StepPerformer.test.ts

```diff
@@ -283,4 +284,38 @@ describe('StepPerformer', () => {
         expect(mockCodeEvaluator.evaluate).toHaveBeenCalledWith('generated code', mockContext);
         expect(mockCacheHandler.addToTemporaryCache).toHaveBeenCalled();
     });
+
+    describe('extendJSContext', () =>{
+        it('should extend the context with the given object', async () => {
+            // Initial context
+            stepPerformer.extendJSContext(dummyBarContext1);
+
+            setupMocks();
+            await stepPerformer.perform(INTENT);
+            expect(mockCodeEvaluator.evaluate).toHaveBeenCalledWith(PROMPT_RESULT, dummyBarContext1);
+
+            // Extended context
+            const newFixedContext = { ...dummyBarContext1, ...dummyContext };
```

Comment:
I think `expectedContext` or `extendedContext` is more semantically correct here